File: Ruby/exercises/bubble_sort.rb
arr = [2, 96, 69, 42, 8, 14]

for i in 0...arr.length do
  for j in 1...arr.length do
    if arr[j - 1] > arr[j]
      temp = arr[j]
      arr[j] = arr[j - 1]
      arr[j - 1] = temp
    end
  end
end

puts arr

File: C++/exercises/bubble-sort.cpp
#include<vector>
#include<iostream>
#include<string>

void BubbleSort(std::vector<int>& vector) {
	for (int i = 0; i < vector.size() - 1; i++) {
		for (int j = 1; j < vector.size() - i; j++) {
			if (vector[j - 1] > vector[j]) {
      			int temp = vector[j];
      			vector[j] = vector[j - 1];
      			vector[j - 1] = temp;
			}
		}
	}
}

int main() {

	std::vector<int> numbers = {2, 26, 4, 97, 14, 8};
	BubbleSort(numbers);
	
	for (int num : numbers) {
	std::cout << num << std::endl;
	}
}
